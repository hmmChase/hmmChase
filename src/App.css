html {
  height: 100%;
}

body {
  min-height: 100%;
}

.App {
  display: flex;
}

.left {
  display: flex;
  justify-content: center;
  width: 50%;
  background: #fcf3ca;
}

.profile {
  position: fixed;
  display: flex;
  flex-direction: column;
  margin-top: 30px;
}

.logo {
  width: 150px;
  height: 150px;
  border: 3px solid #ef7b7b;
  margin: 30px auto;
  background: #c4eada;
  border-top-left-radius: 50%;
  border-bottom-left-radius: 50%;
  border-bottom-right-radius: 50%;
  -moz-transform: scale(1) rotate(-45deg) translateX(0px) translateY(0px)
    skewX(1deg) skewY(1deg);
  -webkit-transform: scale(1) rotate(-45deg) translateX(0px) translateY(0px)
    skewX(1deg) skewY(1deg);
  -o-transform: scale(1) rotate(-45deg) translateX(0px) translateY(0px)
    skewX(1deg) skewY(1deg);
  -ms-transform: scale(1) rotate(-45deg) translateX(0px) translateY(0px)
    skewX(1deg) skewY(1deg);
  transform: scale(1) rotate(-45deg) translateX(0px) translateY(0px) skewX(1deg)
    skewY(1deg);
}

.logo h1 {
  color: #37427f;
  font-family: 'Lobster', cursive;
  font-size: 2.5rem;
  text-align: center;
  -moz-transform: scale(1) rotate(40deg) translateX(0px) translateY(0px)
    skewX(1deg) skewY(1deg);
  -webkit-transform: scale(1) rotate(40deg) translateX(0px) translateY(0px)
    skewX(1deg) skewY(1deg);
  -o-transform: scale(1) rotate(40deg) translateX(0px) translateY(0px)
    skewX(1deg) skewY(1deg);
  -ms-transform: scale(1) rotate(40deg) translateX(0px) translateY(0px)
    skewX(1deg) skewY(1deg);
  transform: scale(1) rotate(40deg) translateX(0px) translateY(0px) skewX(1deg)
    skewY(1deg);
}

.profile h2 {
  margin-top: 0;
  text-align: center;
}

.skill-container {
}

.curly-open::before {
  /* content: '{'; */
  /* font-size: 20rem; */
  /* display: inline-block; */
  /* float: left; */
}

.curly-close::before {
  /* content: '}'; */
  /* font-size: 20rem; */
  /* display: inline-block; */
  /* float: left; */
}

.skills {
  display: flex;
  flex-flow: wrap;
  justify-content: center;
  width: 250px;
  margin: 0 auto;
}

.skill:nth-of-type(odd) {
  width: 150px;
}

.skill:nth-of-type(even) {
  width: 100px;
}

.skill {
  font-size: 1rem;
}

.platforms {
  display: flex;
  justify-content: center;
  font-size: 3rem;
}

.platform-icon {
  margin: 25px 25px 0 25px;
}

.platform-icon:visited {
  color: #000;
}

.right {
  display: flex;
  flex-direction: column;
  align-items: center;
  overflow: auto;
  width: 50%;
  background: #919190;
}

.project-container {
  width: 80%;
  min-width: 300px;
  border-radius: 10px;
  margin-top: 50px;
  background: #fcf3ca;
}

.project-container:last-of-type {
  margin-bottom: 50px;
}

.project-title {
  margin: 10px;
}

.project-img {
  width: 100%;
  border-top: 1px solid darkgray;
  border-bottom: 1px solid darkgray;
}

.project-description {
  padding: 0 20px;
  margin: 20px 0 0 0;
}

.project-link-container {
  display: flex;
  justify-content: space-around;
}

.project-link {
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 10px;
  border: none;
  margin: 20px;
  color: inherit;
  text-decoration: none;
  cursor: pointer;
  background-color: #c4eada;
  box-shadow: 4px 4px 0 0 #ef7b7b;
  -webkit-transition: all 200ms ease;
  -moz-transition: all 200ms ease;
  -o-transition: all 200ms ease;
  transition: all 200ms ease;
}

.project-link:hover {
  transform: translateY(4px) translateX(4px);
  -webkit-transition: all 200ms ease;
  -moz-transition: all 200ms ease;
  -o-transition: all 200ms ease;
  transition: all 200ms ease;
  background-color: #ef7b7b;
  box-shadow: -4px -4px 0 0 #c4eada;
}

@media only screen and (max-width: 700px) {
  .App {
    flex-direction: column;
  }

  .left {
    width: 100%;
    padding-bottom: 20px;
  }

  .profile {
    position: static;
  }

  .right {
    width: 100%;
  }
}
