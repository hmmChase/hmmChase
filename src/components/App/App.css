.App {
  background: #fcf3ca;
}

.left {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 4rem 0 .5rem 0;
}

.fixed {
  display: flex;
  flex-direction: column;
  align-items: center;
  padding-bottom: 1.5rem;
}

.logo {
  width: 150px;
  height: 150px;
  border: 3px solid #ef7b7b;
  border-radius: 50% 0 50% 50%;
  background: #c4eada;
  -moz-transform: scale(1) rotate(-45deg) translateX(0px) translateY(0px)
    skewX(1deg) skewY(1deg);
  -webkit-transform: scale(1) rotate(-45deg) translateX(0px) translateY(0px)
    skewX(1deg) skewY(1deg);
  -o-transform: scale(1) rotate(-45deg) translateX(0px) translateY(0px)
    skewX(1deg) skewY(1deg);
  -ms-transform: scale(1) rotate(-45deg) translateX(0px) translateY(0px)
    skewX(1deg) skewY(1deg);
  transform: scale(1) rotate(-45deg) translateX(0px) translateY(0px) skewX(1deg)
    skewY(1deg);
}

.logo h1 {
  font-family: 'Lobster', cursive;
  font-size: 2.5rem;
  color: #37427f;
  text-align: center;
  -moz-transform: scale(1) rotate(40deg) translateX(0px) translateY(0px)
    skewX(1deg) skewY(1deg);
  -webkit-transform: scale(1) rotate(40deg) translateX(0px) translateY(0px)
    skewX(1deg) skewY(1deg);
  -o-transform: scale(1) rotate(40deg) translateX(0px) translateY(0px)
    skewX(1deg) skewY(1deg);
  -ms-transform: scale(1) rotate(40deg) translateX(0px) translateY(0px)
    skewX(1deg) skewY(1deg);
  transform: scale(1) rotate(40deg) translateX(0px) translateY(0px) skewX(1deg)
    skewY(1deg);
}

.projects-heading {
  font-size: 2.5rem;
}

.projects-label {
  color: #37427f;
}

.levelDown {
  margin: 0 1rem;
  font-size: 1.5rem;
}

.right {
  background: #919190;
}

@media only screen and (min-width: 700px) {
  .App {
    display: flex;
  }

  .left {
    position: relative;
    align-items: center;
    width: 50%;
    height: 100vh;
    padding: 0;
  }

  .fixed {
    position: fixed;
  }

  .projects-heading {
    position: fixed;
    right: calc(50% - 130px);
    font-size: 3rem;
    transform: rotate(-90deg);
  }

  .levelDown {
    margin: 0 1rem;
    font-size: 2rem;
  }

  .right {
    width: 50%;
  }
}
