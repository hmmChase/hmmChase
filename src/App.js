import React, { Component } from 'react';
import './App.css';

class App extends Component {
  render() {
    return (
      <main className="App">
        <section className="left">left</section>
        <section className="right">right</section>
      </main>
    );
  }
}

export default App;
