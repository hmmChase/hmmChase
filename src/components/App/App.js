import React from 'react';
import { library } from '@fortawesome/fontawesome-svg-core';
import { fab } from '@fortawesome/free-brands-svg-icons';
import Profile from '../Profile/Profile';
import ProjectContainer from '../ProjectContainer/ProjectContainer';
import './App.css';

library.add(fab);

const App = () => (
  <main className='App'>
    <section className='left'>
      <div className='fixed'>
        <div className='logo'>
          <h1>hmm... Chase</h1>
        </div>

        <Profile />
      </div>
    </section>

    <section className='right'>
      <ProjectContainer />
    </section>
  </main>
);

export default App;
