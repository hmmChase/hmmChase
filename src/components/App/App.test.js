import React from 'react';
import ReactDOM from 'react-dom';
import { render, prettyDOM } from '@testing-library/react';
import App from './App';

describe('App', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('matches snapshot', () => {
    const result = render(<App />);

    // console.log('TCL: result', result);

    console.log('TCL: result', result.debug());

    expect(result.asFragment()).toMatchSnapshot();
  });

  it('has a logo', () => {
    const result = render(<App />);

    const logo = result.getByTestId('logo');

    // console.log(prettyDOM(logo));

    expect(logo).toHaveTextContent('hmm... Chase');
  });
});
