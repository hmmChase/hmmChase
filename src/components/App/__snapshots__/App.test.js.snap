// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
<DocumentFragment>
  <main
    class="App"
  >
    <section
      class="left"
    >
      <div
        class="fixed"
      >
        <div
          class="logo"
        >
          <h1
            data-testid="logo"
          >
            hmm... Chase
          </h1>
        </div>
        <div>
          <h2>
            Web Developer
          </h2>
          <div
            class="skills"
          >
            <div
              class="skills-left"
            >
              <span
                class="skill"
              >
                JavaScript
              </span>
              <span
                class="skill"
              >
                jQuery
              </span>
              <span
                class="skill"
              >
                HTML
              </span>
              <span
                class="skill"
              >
                CSS/SCSS
              </span>
              <span
                class="skill"
              >
                Git
              </span>
              <span
                class="skill"
              >
                React
              </span>
              <span
                class="skill"
              >
                Router
              </span>
              <span
                class="skill"
              >
                Redux
              </span>
              <a
                class="platform-icon"
                href="https://github.com/hmmChase"
              >
                <i
                  class="fa fab-github undefined"
                />
              </a>
            </div>
            <hr
              class="split"
            />
            <div
              class="skills-right"
            >
              <span
                class="skill"
              >
                NextJS
              </span>
              <span
                class="skill"
              >
                Express
              </span>
              <span
                class="skill"
              >
                GraphQL
              </span>
              <span
                class="skill"
              >
                Postgres
              </span>
              <span
                class="skill"
              >
                Mocha
              </span>
              <span
                class="skill"
              >
                Chai
              </span>
              <span
                class="skill"
              >
                Jest
              </span>
              <span
                class="skill"
              >
                Enzyme
              </span>
              <a
                class="platform-icon"
                href="https://www.linkedin.com/in/chase-r/"
              >
                <i
                  class="fa fab-linkedin undefined"
                />
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="projects-heading"
      >
        <i
          class="fa fas-level-down-alt undefined"
        />
        <span>
          projects
        </span>
        <i
          class="fa fas-level-down-alt undefined"
        />
      </div>
    </section>
    <section
      class="right"
    >
      <div
        class="project-container"
      >
        <article
          class="project"
        >
          <h3
            class="project-title"
          >
            IdeaBox
          </h3>
          <img
            alt="screenshot"
            class="project-img"
            src="ideabox.png"
          />
          <p
            class="project-description"
          >
            IdeaBox is a simple Create, Read, Update, Delete (CRUD) app.  It is implemented three different ways.
          </p>
          <div
            class="project-link-container"
          >
            <div>
              <p
                class="project-link-name"
              >
                v1. Vanilla JS
              </p>
              <div
                class="project-links"
              >
                <a
                  class="project-link"
                  href="https://github.com/hmmChase/ideabox-v1"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  GitHub
                </a>
                <a
                  class="project-link"
                  href="https://hmmchase.github.io/ideabox-v1/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  View Live
                </a>
              </div>
            </div>
            <div>
              <p
                class="project-link-name"
              >
                v2. React / RESTful
              </p>
              <div
                class="project-links"
              >
                <a
                  class="project-link"
                  href="https://github.com/hmmChase/ideabox-v2"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  GitHub
                </a>
                <a
                  class="project-link"
                  href="https://ideabox-cra-rest.herokuapp.com/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  View Live
                </a>
              </div>
            </div>
            <div>
              <p
                class="project-link-name"
              >
                v3. SSR React / GraphQL
              </p>
              <div
                class="project-links"
              >
                <a
                  class="project-link"
                  href="https://github.com/hmmChase/ideabox-v3"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  GitHub
                </a>
                <a
                  class="project-link"
                  href="https://ideabox-v3.hmmchase.now.sh/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  View Live
                </a>
              </div>
            </div>
            <div>
              <p
                class="project-link-name"
              >
                v4. v3 + user accounts
              </p>
              <div
                class="project-links"
              >
                <a
                  class="project-link"
                  href="https://github.com/hmmChase/ideabox-v4"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  GitHub
                </a>
                <a
                  class="project-link"
                  href="https://ideabox-v4.hmmchase.now.sh/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  View Live
                </a>
              </div>
            </div>
          </div>
        </article>
        <article
          class="project"
        >
          <h3
            class="project-title"
          >
            BitPoll
          </h3>
          <img
            alt="screenshot"
            class="project-img"
            src="bitpoll.png"
          />
          <p
            class="project-description"
          >
            Bitpoll helps determine consensus within the Bitcoin development community by providing polls restricted to developers of the bitcoin project.
          </p>
          <div
            class="project-link-container"
          >
            <div>
              <div
                class="project-links"
              >
                <a
                  class="project-link"
                  href="https://github.com/hmmChase/bitpoll"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  GitHub
                </a>
                <a
                  class="project-link"
                  href="https://bit-poll.herokuapp.com/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  View Live
                </a>
              </div>
            </div>
          </div>
        </article>
        <article
          class="project"
        >
          <h3
            class="project-title"
          >
            Monikers
          </h3>
          <img
            alt="screenshot"
            class="project-img"
            src="monikers.png"
          />
          <p
            class="project-description"
          >
            A mobile version of the card game Monikers.
          </p>
          <div
            class="project-link-container"
          >
            <div>
              <div
                class="project-links"
              >
                <a
                  class="project-link"
                  href="https://github.com/hmmChase/monikers"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  GitHub
                </a>
                <a
                  class="project-link"
                  href="https://monikers.herokuapp.com"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  View Live
                </a>
              </div>
            </div>
          </div>
        </article>
        <article
          class="project"
        >
          <h3
            class="project-title"
          >
            Palette Picker
          </h3>
          <img
            alt="screenshot"
            class="project-img"
            src="palette-picker.png"
          />
          <p
            class="project-description"
          >
            Palette Picker allows you to generate color schemes and save them in project folders.
          </p>
          <div
            class="project-link-container"
          >
            <div>
              <div
                class="project-links"
              >
                <a
                  class="project-link"
                  href="https://github.com/hmmChase/palette-picker"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  GitHub
                </a>
                <a
                  class="project-link"
                  href="https://hmm-palette-picker.herokuapp.com/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  View Live
                </a>
              </div>
            </div>
          </div>
        </article>
        <article
          class="project"
        >
          <h3
            class="project-title"
          >
            hmm... Chase
          </h3>
          <img
            alt="screenshot"
            class="project-img"
            src="hmmchase.png"
          />
          <p
            class="project-description"
          >
            You are here.
          </p>
          <div
            class="project-link-container"
          >
            <div>
              <div
                class="project-links"
              >
                <a
                  class="project-link"
                  href="https://github.com/hmmChase/hmmChase"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  GitHub
                </a>
                <a
                  class="project-link"
                  href="https://www.youtube.com/watch?v=mz6tAJMVmfM"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  View Live
                </a>
              </div>
            </div>
          </div>
        </article>
      </div>
    </section>
  </main>
</DocumentFragment>
`;
